reports-backend/src/main/java/org/jboss/da/listings/api/dao/ProductDAO.java

```diff
@@ -0,0 +1,100 @@
+package org.jboss.da.listings.api.dao;
+
+import org.jboss.da.listings.api.model.Product;
+import org.jboss.da.listings.api.model.ProductArtifactRelationship;
+import org.jboss.da.listings.api.service.ArtifactService.SupportStatus;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+public interface ProductDAO extends GenericDAO<Product> {
+
+    /**
+     * Change support status of Product with specific name and version
+     * 
+     * @param name
+     * @param version
+     * @param newStatus
+     * @return True if change was successful otherwise false
+     * @throws NoSuchElementException
+     */
+    boolean changeProductStatus(String name, String version, SupportStatus newStatus)
+            throws NoSuchElementException;
+
+    /**
+     * Finds Artifacts of Product with specific name and version
+     * 
+     * @param name
+     * @param version
+     * @return Product with its artifacts
+     */
+    Product findArtifactsOfProduct(String name, String version);
```

Comment:
I use method that returns Optional, this method is never used. I'll check why i created this and remove it if there is no point